modules/services/home-manager-auto-upgrade.nix

```diff
@@ -37,6 +62,9 @@ in {
         lib.platforms.linux)
     ];
 
+    services.home-manager.autoUpgrade.flags =
+      if cfg.flake != null then [ "--flake" cfg.flake ] else [ ];
```

Comment:
```suggestion
      lib.optionals (cfg.flake != null) [ "--flake" cfg.flake ];
```

Reply:
Why not, I mean there are many way to do it, I was inspired by the Nixpkgs' own implementation of the feature.